plugins/bootc.py

```diff
@@ -0,0 +1,125 @@
+            self.opts.subcmd = self._BOOTC_ALIASES[self.opts.subcmd]
+
+    def __init__(self, cli):
+        super().__init__(cli)
+
+    @staticmethod
+    def set_argparser(parser):
+        # subcommands for the plugin
+        parser.add_argument(
+            "subcmd",
+            nargs="?",
+            metavar="BOOTC",
+            help=_("available subcommands: {} (default), {}").format(
+                BootcCommand._BOOTC_SUBCOMMANDS[0],
+                ", ".join(BootcCommand._BOOTC_SUBCOMMANDS[1:]),
+            ),
+        )
+
+        # these options are for 'status'
+        parser.add_argument(
+            "--json", action="store_true", help=_("Output JSON (status)")
+        )
+        parser.add_argument(
+            "--booted",
+            action="store_true",
+            help=_("Only print the booted deployment (status)"),
+        )
+        parser.add_argument(
+            "--jsonpath",
+            metavar="EXPRESSION",
+            action="store",
+            help=_("Filter JSONPath expression (status)"),
+        )
+        parser.add_argument(
+            "--pending-exit-77",
+            action="store_true",
+            help=_("If pending deployment available, exit 77 (status)"),
+        )
+        parser.add_argument(
+            "--peer",
+            action="store_true",
+            help=_(
+                "Force a peer-to-peer connection instead of using the system message bus (status)"
+            ),
+        )
+
+    def configure(self):
+        super().configure()
+
+        self._canonical()
+        cmd = self.opts.subcmd
+
+        # ensure we have a valid subcommand
+        if cmd not in self._BOOTC_SUBCOMMANDS:
+            logger.critical(
+                _("Invalid bootc sub-command, use: %s."),
+                ", ".join(self._BOOTC_SUBCOMMANDS),
+            )
+            raise dnf.cli.CliError
+
+        self.extargs = [self._EXT_CMD, cmd]
+
+        # process subcommand arguments
+        if cmd == "status":
+            if self.opts.quiet:
+                self.extargs.append("-q")
+            elif self.opts.verbose:
+                self.extargs.append("-v")
+            elif self.opts.json:
+                self.extargs.append("--json")
+            elif self.opts.jsonpath:
+                self.extargs.append("--jsonpath=%s" % self.opts.jsonpath)
+            elif self.opts.booted:
+                self.extargs.append("-b")
+            elif self.opts.pending_exit_77:
+                self.extargs.append("--pending-exit-77")
+            elif self.opts.peer:
+                self.extargs.append("--peer")
+            elif self.opts.installroot:
+                self.extargs.append("--sysroot=%s" % self.opts.installroot)
+
+    def run(self):
+        proc = subprocess.Popen(
+            self.extargs, stdout=subprocess.PIPE, stderr=subprocess.STDOUT
+        )
+        output = proc.communicate()
+
+        if proc.returncode != 0:
+            logger.critical(output)
```

Comment:
I made a commit [here](https://github.com/evan-goode/dnf-plugins-core/tree/evan-goode/bootc) to fix the output, currently it is printing the `__str__()` of the stdout, stderr tuple which looks like `(b"stderr\n", b"stdout\n") and is not pretty.

In general, we will want better logic here to display the output from rpm-ostree that preserves the order of lines logged to stdout and stderr (versus logging all stdout, then all stderr). But this is alright for now; the `status` subcommand is pretty simple.  Ideally, colors from rpm-ostree's output would be displayed too, but that might be awkward since colored output shouldn't go in logs. And it doesn't seem `rpm-ostree` has a `--color` flag to force colors on and off, it's probably checking whether its stdout is a terminal.

Reply:
> I made a commit [here](https://github.com/evan-goode/dnf-plugins-core/tree/evan-goode/bootc) to fix the output, currently it is printing the `__str__()` of the stdout, stderr tuple which looks like `(b"stderr\n", b"stdout\n")` and is not pretty.

Oh yeah, gross.  I was thinking sending stderr to STDOUT would keep the output interleaved as it would display on the terminal, which would be suitable for logging.

> In general, we will want better logic here to display the output from rpm-ostree that preserves the order of lines logged to stdout and stderr (versus logging all stdout, then all stderr). But this is alright for now; the `status` subcommand is pretty simple. Ideally, colors from rpm-ostree's output would be displayed too, but that might be awkward since colored output shouldn't go in logs. And it doesn't seem `rpm-ostree` has a `--color` flag to force colors on and off, it's probably checking whether its stdout is a terminal.

Agreed, this is a part of the plugin I envision evolving as more commands are added.  Color may be a reach, but probably not impossible.